Eleventy Bookshop Guide (#55)
CloudCannon/platform-documentation: guides/bookshop-eleventy-guide/getting-set-up.mdx

description: "The first steps for building an Eleventy site with Bookshop"
tags: []
related_articles: null
related_links: []
---

Building a site with Bookshop requires a specific directory structure that lives alongside your website,
and an Eleventy plugin to discover and use components.

<comp.Notice info_type="info">
   As a useful reference, the [Eleventy Bookshop Starter](https://github.com/CloudCannon/eleventy-bookshop-starter/) repository 
   contains a simple site fully integrated with visual editing. Browsing the code and the file structure
   of the starter site can be helpful to solidify the concepts covered in this guide.
</comp.Notice>

## Prerequisites

Bookshop requires Node >= 16 to be installed on your machine.

## Creating your Bookshop directory

Bookshop requires that your component files live in a specific directory structure, which allows for components
to be discovered automatically. 

The easiest way to get started is to run Bookshop's `init` command. This should be run alongside your website source files:

<comp.CodeBlock language="bash" source="Command line">
```
npx @bookshop/init --new _component-library --framework eleventy
```
</comp.CodeBlock>

> After running this command, the following directory structure should have been created:

Reviewer: ```suggestion
After running this command, the following directory structure will be created:
```
